handler/src/main/java/io/netty/handler/stream/ChunkedWriteHandler.java

```diff
@@ -209,6 +209,18 @@ private void doFlush(final ChannelHandlerContext ctx) {
             if (currentWrite == null) {
                 break;
             }
+
+            if (currentWrite.promise.isDone()) {
+                // This might happen e.g. in the case when a write operation
+                // failed, but there're still unconsumed chunks left.
+                // Most chunked input sources would stop generating chunks
+                // and report end of input, but this doesn't work with any
+                // source wrapped in HttpChunkedInput.
+                // See https://github.com/netty/netty/issues/8700.
+                this.currentWrite = null;
```

Comment:
do we also need to release the message ?

Reply:
Not here... I assume someone who resolved the promise (either with success or with failure) have already done this by calling `closeInput`. As we're not allocating any new chunks (calling `readChunk`), everything else should be the responsibility of a specific input implementation.

Reply:
good point. Maybe add a comment ?

Reply:
Will do!